packages/zoe/src/zoeService/zoe.js

```diff
@@ -44,37 +44,87 @@ import { ZoeServiceIKit } from '../typeGuards.js';
  * @param {Baggage} [zoeBaggage]
  */
 const makeZoeKit = (
-  vatAdminSvc,
+  vatAdminSvcP = undefined,
   shutdownZoeVat = () => {},
   feeIssuerConfig = defaultFeeIssuerConfig,
   zcfSpec = { name: 'zcf' },
   zoeBaggage = makeScalarBigMapStore('zoe baggage', { durable: true }),
 ) => {
-  let zoeService;
+  let zcfBundleCap;
+
+  function saveBundleCap() {
+    E.when(
+      Promise.all([vatAdminSvcP, getZcfBundleCap(zcfSpec, vatAdminSvcP)]),
+      ([vatAdminService, bundleCap]) => {
+        zcfBundleCap = bundleCap;
+
+        zoeBaggage.init('vatAdminSvc', vatAdminService);
+        zoeBaggage.init('zcfBundleCap', zcfBundleCap);
+      },
+    );
+  }
+
+  const setVatAdminService = Far('setVatAdminService', lateVatAdminSvc => {
+    vatAdminSvcP = lateVatAdminSvc;
+    saveBundleCap();
+  });
```

Comment:
I don't believe so. 

`setVatAdminService` is only available to the party that created Zoe itself. They have many other ways they can break Zoe, so the question is whether it's hard to use correctly. The only useful time to call it is in a vat that is starting Zoe and wanting to support the possibility of that Zoe being restarted.

If it were called again, Zoe would throw an exception because of the repeated `zoeBaggage.init()` calls in `saveBundleCap()` and crash. I think whoever created Zoe can cause the same thing to happen in simpler ways. 

If you think this deserves a guard or a comment explaining this, I'm willing to add them, but it doesn't seem like a point of vulnerability to me.